integrations/helm/chartsync/chartsync.go

```diff
@@ -339,22 +339,22 @@ func (chs *ChartChangeSync) reconcileReleaseDef(fhr fluxv1beta1.HelmRelease) {
 		chartSource := fhr.Spec.ChartSource.RepoChartSource
 		path, err := ensureChartFetched(chs.config.ChartCache, chartSource)
 		if err != nil {
-			chs.setCondition(&fhr, fluxv1beta1.HelmReleaseChartFetched, v1.ConditionFalse, ReasonDownloadFailed, "chart download failed: "+err.Error())
+			chs.setCondition(&fhr, fluxv1beta1.HelmReleaseChartFetched, v1.ConditionFalse, ReasonDownloadFailed, "chart download failed: "+err.Error(), refHead)
```

Comment:
It doesn't make much sense to put the refHead here (or on L346), since the condition pertains to downloading a chart package from a helm repo

Reply:
I agree, I didn't like this approach... That was a quick POC for myself. I have taken a different simpler approach (force pushed).